Assets/Scripts/UI/PaletteSwap.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Netherlands3D.Twin
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEditor;
+    using UnityEngine;
+    using UnityEngine.UI;
+
+    public class PaletteSwap : MonoBehaviour
+    {
+        private Image image;
+        // [SerializeField] private List<Color> colors;
+        [SerializeField] private ColorPalette palette;
+        [SerializeField] private bool allowRuntimeUpdateChange;
+
+        private static readonly int PaletteTex = Shader.PropertyToID("_PaletteTex");
+
+        private void Awake()
+        {
+            UpdatePalette();
+        }
+
+        private void OnValidate()
+        {
+            image = GetComponent<Image>();
+        
+            UpdatePalette();
+        }
+
+        private void Update()
+        {
+            if (!allowRuntimeUpdateChange)
+                return;
+        
+            UpdatePalette();
+        }
+
+        public void UpdatePalette()
+        { 
+            image.material.SetTexture(PaletteTex, GetTexture());
+        }
+
+        private Texture2D GetTexture()
+        {
+            Texture2D newTexture = new Texture2D(palette.Colors.Count, 1, TextureFormat.RGBA32, false);
```

Comment:
Gaat het geheugengebruik niet langzaamaan oplopen als er elke keer een nieuwe texture wordt gemaakt en de oude niet vrijgegeven? Ik zie dat je setpixel gebruikt, kan dat niet op een bestaande texture; dat je die in Awake ofzo cloned?

Reply:
elke instance van dit script maakt een palette texture ja, maar dit is dan een texture van 1x16 pixels ongeveer, dus dat vond ik een acceptabel geheugengebruik. 

als je weer GetTexture gebruikt tijdens runtime neem ik aan dat de garbage collector de oude weer opruimt.